var/spack/repos/builtin/packages/spdk/package.py

```diff
@@ -0,0 +1,106 @@
+    url = "https://github.com/spdk/spdk/archive/v23.01.tar.gz"
+    git = "https://github.com/spdk/spdk"
+    maintainers("hyoklee")
+
+    version("master", branch="master", submodules=True)
+    version("23.01", tag="v23.01", submodules=True)
+
+    variant("crypto", default=False, description="Build vbdev crypto module")
+    variant("dpdk", default=False, description="Build with dpdk")
+    variant("fio", default=False, description="Build fio plugin")
+    variant("iscsi-initiator", default=False, description="Build with iscsi bdev module")
+    variant("ocf", default=False, description="Build OCF library and bdev module")
+    variant("pmdk", default=False, description="Build persistent memory bdev")
+    variant("rbd", default=False, description="Build Ceph RBD bdev module")
+    variant(
+        "rdma", default=False, description="Build RDMA transport for NVMf target and initiator"
+    )
+    variant("shared", default=False, description="Build spdk shared libraries")
+    variant("uring", default=False, description="Build I/O uring bdev")
+    variant(
+        "virtio", default=False, description="Build vhost initiator and virtio-pci bdev modules"
+    )
+    variant("vhost", default=False, description="Build vhost target")
+    variant("vtune", default=False, description="Profile I/O under Intel VTune Amplifier XE")
+
+    mods = (
+        "crypto",
+        "dpdk",
+        "iscsi-initiator",
+        "ocf",
+        "pmdk",
+        "rbd",
+        "rdma",
+        "shared",
+        "uring",
+        "vhost",
+        "virtio",
+        "vtune",
+    )
+
+    depends_on("dpdk@22.11:", when="+dpdk")
+    depends_on("fio@3.33", when="+fio")
+    depends_on("libaio")
+    depends_on("meson")
+    depends_on("nasm@2.12.02:", type="build")
+    depends_on("numactl")
+    depends_on("py-pyelftools")
+    depends_on("rdma-core", when="+rdma")
+
+    def configure_args(self):
+        spec = self.spec
+        config_args = ["--disable-tests", "--disable-unit-tests", "--disable-apps"]
+
+        if "+fio" in spec:
+            config_args.append("--with-fio={0}".format(spec["fio"].prefix))
+
+        for mod in self.mods:
+            if "+" + mod in spec:
+                config_args.append("--with-{0}".format(mod))
+            else:
+                config_args.append("--without-{0}".format(mod))
+
+        return config_args
+
+    @run_after("install")
+    def install_additional_files(self):
+        spec = self.spec
+        prefix = self.prefix
+        dpdk_build_dir = join_path(spec["dpdk"].prefix, "lib")
+
+        install_tree(join_path(dpdk_build_dir, "pkgconfig"), join_path(prefix.lib, "pkgconfig"))
+
+        for file in os.listdir(dpdk_build_dir):
+            f = join_path(dpdk_build_dir, file)
+            if os.path.isfile(f):
+                install(f, prefix.lib)
+
+        mkdir(join_path(prefix.include, "dpdk"))
+        install_tree(join_path(spec["dpdk"].prefix, "include"), join_path(prefix.include, "dpdk"))
```

Comment:
here you should have:
```
if "+dpdk" in spec:
        dpdk_build_dir = join_path(spec["dpdk"].prefix, "lib")
        install_tree(join_path(dpdk_build_dir, "pkgconfig"), join_path(prefix.lib, "pkgconfig"))
        for file in os.listdir(dpdk_build_dir):
            f = join_path(dpdk_build_dir, file)
            if os.path.isfile(f):
                install(f, prefix.lib)
        mkdir(join_path(prefix.include, "dpdk"))
        install_tree(join_path(spec["dpdk"].prefix, "include"), join_path(prefix.include, "dpdk"))
```
that will expose the internal dpdk install from spdk so that we are using the same version.